app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb

```diff
@@ -161,4 +176,22 @@ def raw_detach_volume(vm_ems_ref)
     _log.error("volume=[#{name}], error: #{e}")
     raise MiqException::MiqVolumeDetachError, _("Unable to detach volume: %{error_message}") % {:error_message => e.message}
   end
+
+  def raw_clone_volume(options)
+    options[:volume_i_ds] = [ems_ref]
+    with_provider_connection(:service => 'PCloudVolumesApi') do |api|
+      clone_volume_params = IbmCloudPower::VolumesCloneAsyncRequest.new(
+        :name        => options['name'],
+        :volume_i_ds => options[:volume_i_ds]
+      )
+      api.pcloud_v2_volumes_clone_post(
+        cloud_instance_id,
+        clone_volume_params
+      )
+    end
+  rescue => e
+    _log.error("volume=[#{name}], error: #{e}")
+    # Raise Correct Error
+    raise MiqException::MiqVolumeDeleteError, e.to_s, e.backtrace
```

Comment:
Either need to define a `MiqException::MiqVolumeCloneError` class or raise something more generic.

Reply:
I think this is where you would add it:
https://github.com/ManageIQ/manageiq-gems-pending/blob/master/lib/gems/pending/util/miq-exception.rb#L119